src/components/Search/SelectedRevisionItem.tsx

```diff
@@ -65,6 +67,13 @@ function SelectedRevisionItem({
   const selectedFramework = useAppSelector((state) => state.framework);
   const navigate = useNavigate();
   const prevRevRef = React.useRef<RevisionsList[]>([]);
+  const location = useLocation();
+  const view = location.pathname == '/' ? searchView : compareView;
+  //hide the close icon for the selected revisions in compare view
+  const iconClassName =
+    !formIsDisplayed && view !== searchView
```

Comment:
We'll have to think a bit about how to remove this remaning condition using the view, I'm sure we can come up with something smart.